### PR TITLE
RARTPAD : add support via

### PR DESCRIPTION
**Pull request by ladbahuy**
## Description

add via support and add reset for default and numpad keymaps

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rart/rartpad/keymaps/numpad/keymap.c [hunk 1]
```diff
@@ -24,7 +24,14 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
         KC_P1,   KC_P2,   KC_P3,
-        KC_P0,            KC_PDOT, KC_PENT
+        KC_P0,            MO(1),   KC_PENT
+    ),
+    [1] = LAYOUT_numpad_5x4(
+        KC_TRNS, KC_TRNS, KC_TRNS,  RESET, 
+        KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS,          KC_PDOT,  KC_TRNS
```
Comment (fauxpark) [inline keyboards/rart/rartpad/keymaps/numpad/keymap.c:34]:
```suggestion
        KC_TRNS,          KC_TRNS,  KC_PDOT 
```

File: keyboards/rart/rartpad/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  [0] = LAYOUT_ortho_5x4(
+    KC_NLCK, KC_PSLS, KC_PAST, KC_INS,
+    KC_P7,   KC_P8,   KC_P9,   KC_PEQL,
+    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+    KC_P1,   KC_P2,   KC_P3,   KC_PMNS,
+    MO(1),   KC_P0,   KC_PDOT, KC_PENT
+    ),
+ 
+  [1] = LAYOUT_ortho_5x4(
+    KC_TRNS, RGB_HUI, RGB_HUD,  KC_TRNS,
```
Comment (fauxpark) [inline keyboards/rart/rartpad/keymaps/via/keymap.c:14]:
```suggestion
    KC_TRNS, RGB_HUI, RGB_HUD,  RESET, 
```

File: keyboards/rart/rartpad/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  [1] = LAYOUT_ortho_5x4(
+    KC_TRNS, RGB_HUI, RGB_HUD,  KC_TRNS, 
+    RGB_SAI, RGB_SAD, KC_MNXT,  KC_MPRV, 
+    RGB_VAI, RGB_VAD, KC_MSTP,  KC_MPLY, 
+    KC_COPY, KC_PSTE, KC_MYCM,  KC_CALC, 
+    KC_TRNS, RGB_TOG, KC_TRNS,  KC_TRNS 
+    ),
+    
+  [2] = LAYOUT_ortho_5x4(
+    _______, _______, _______, RESET,
```
Comment (fauxpark) [inline keyboards/rart/rartpad/keymaps/via/keymap.c:22]:
```suggestion
    _______, _______, _______, _______,
```
There's no way to get to this layer.

Comment (ladbahuy) [reply keyboards/rart/rartpad/keymaps/via/keymap.c]: @fauxpark oke done fix it

**Comment by ladbahuy** [comment]
hii @fauxpark can you give an example of using a rotary encoder on via?

**Comment by fauxpark** [comment]
VIA currently doesn't support rotary encoders, except as hardcoded into the keymap.

**Comment by ladbahuy** [comment]
@fauxpark okee,  thank you

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**